contracts/voting/Voting.sol

```diff
@@ -0,0 +1,350 @@
+// SPDX-License-Identifier: Apache 2.0
+// Copyright © 2021 Anton "BaldyAsh" Grigorev. All rights reserved.
+
+pragma solidity ^0.8.0;
+
+import "../common/ownership/Ownable.sol";
+import "../nft/NFT.sol";
+import "../vote/Vote.sol";
+import "../voucher/Voucher.sol";
+import "./IVoting.sol";
+import "./VotingConstants.sol";
+import "./VotingStructs.sol";
+import "./VotingErrors.sol";
+
+contract Voting is IVoting, Ownable {
+    uint256 internal constant VOTING_DURATION = 1 days;
+    uint256 internal constant GOVERNANCE_DURATION = 6 days;
+
+    uint256 internal constant PRIZE_RATE = 87;
+    uint256 internal constant REWARD_BURN_RATE = 3;
+
+    NFT internal _mayor;
+    Vote internal _voteToken;
+    Voucher internal _voucherToken;
+
+    // amount of votes for 1 citizen, 4 digits
+    uint256 internal _votesPerCitizen;
+    uint256 internal _voteDigits;
+
+    // city id => City
+    mapping(uint256 => City) internal _cities;
+    // region id => Region
+    mapping(uint256 => Region) internal _regions;
+    // regionId => [city id]
+    mapping(uint256 => uint256[]) internal _regionToCities;
+
+
+    // TODO: change bool to the timestamp for calculating Monument reward?
+    // owner address => city id => Building => 0: not exists, >0: season number +1
+    mapping(address => mapping(uint256 => mapping(Building => uint256))) internal _ownerToBuildings;
+
+    // owner address => city id => season => claimed
+    mapping(address => mapping(uint256 => mapping(uint256 => bool))) internal _ownerClaims;
+
+    // city id => season id => Nominee[]
+    mapping(uint256 => mapping(uint256 => Nominee[])) internal _cityToNominees;
+
+    uint256 internal _cityIdCounter = 0;
+
+    constructor(
+        NFT mayorNFT,
+        Vote voteToken_,
+        Voucher voucherToken_,
+        uint256 votesPerCitizen_,
+        address owner
+    ) {
+        _mayor = mayorNFT;
+        _voteToken = voteToken_;
+        _voucherToken = voucherToken_;
+        _votesPerCitizen = votesPerCitizen_;
+        _owner = owner;
+        _voteDigits = _voteToken.decimals();
+    }
+
+    function transferTokens(address recipient) external override isOwner {
```

Comment:
unsafe. you can transfer tokens from unclaimed votings. propose to create fees variables and withdraw only them

Reply:
I think the problem was solved, but you should read my comments to you in slack about that